src/PlaywrightSharp/BrowserContext.cs

```diff
@@ -239,6 +239,12 @@ public async Task<T> WaitForEventAsync<T>(PlaywrightEvent<T> e, Func<T, bool> pr
                 waiter.RejectOnEvent<EventArgs>(this, ContextEvent.Close.Name, new TargetClosedException("Context closed"));
             }
 
+            if (typeof(T) == typeof(EventArgs))
```

Comment:
This will stay I think. Events like `Close` will be  `PlaywrightEvent<EventArgs>` so that would be a hit for us so we don't return any event args.  I created an issue https://github.com/microsoft/playwright-sharp/issues/1276 so we can review if we can kill this.